### PR TITLE
May Funding Update Part B

### PR DESCRIPTION
**Pull request by defijesus**

### CODE REVIEW

File: src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/AaveV3Ethereum_MayFundingUpdatePartB_20240917.sol [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      aToken: AaveV2EthereumAssets.WETH_A_TOKEN,
+      leaveBehind: 1e18
+    });
+
+    for (uint i = 0; i < migrations.length; i++) {
+      /// transfer aToken
+      AaveV3Ethereum.COLLECTOR.transfer(
+        migrations[i].aToken,
+        address(this),
+        IScaledBalanceToken(migrations[i].aToken).scaledBalanceOf(COLLECTOR) -
```
Comment (sakulstra) [inline src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/AaveV3Ethereum_MayFundingUpdatePartB_20240917.sol:56]: why using scaledBalance here? seems to be wrong imo (at least different to what you write in the aip)

File: src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/AaveV3Ethereum_MayFundingUpdatePartB_20240917.sol [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      aToken: AaveV2EthereumAssets.USDC_A_TOKEN,
+      leaveBehind: 1e6
+    });
+    migrations[3] = Migration({
+      underlying: AaveV2EthereumAssets.WETH_UNDERLYING,
+      aToken: AaveV2EthereumAssets.WETH_A_TOKEN,
+      leaveBehind: 1e18
+    });
+
+    for (uint i = 0; i < migrations.length; i++) {
```
Comment (sakulstra) [inline src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/AaveV3Ethereum_MayFundingUpdatePartB_20240917.sol:51]: would you mind migrating to using https://github.com/bgd-labs/aave-helpers/blob/master/src/CollectorUtils.sol

In this case you would:
- balance = withdrawFromV2();
- depositToV3();

and as `amount` you specify `balanceOf(collector)-margin`

File: src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/MayFundingUpdatePartB.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Simple Summary
+
+This AIP will deposit idle assets and migrate v2 treasury holdings to Aave v3 on Ethereum. This AIP also executes the [Request for Bounty Payout - September 2024](https://governance.aave.com/t/bgd-request-for-bounty-payout-september-2024/19256).
+
+## Motivation
+
+This AIP furthers the ongoing migration of Aave DAO Treasury assets from v2 to v3. Additionally, it proposes depositing assets currently not generating yield, which are held in the Treasury from previous funding proposals, into Aave v3 on Ethereum.
+
+## Specification
```
Comment (eboadom) [inline src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/MayFundingUpdatePartB.md:16]: Better to add in the specification the details of the bug bounty transfers

File: src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/AaveV3Ethereum_MayFundingUpdatePartB_20240917.sol [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+ */
+contract AaveV3Ethereum_MayFundingUpdatePartB_20240917 is IProposalGenericExecutor {
+  using SafeERC20 for IERC20;
+
+  address public constant COLLECTOR = address(AaveV3Ethereum.COLLECTOR);
+  address public constant POOL_V2 = address(AaveV2Ethereum.POOL);
+  address public constant POOL_V3 = address(AaveV3Ethereum.POOL);
+  address public constant RECIPIENT_500 = 0x1770776deB0A5CA58439759FAdb5cAA014501241;
+  address public constant RECIPIENT_1000 = 0x7dF98A6e1895fd247aD4e75B8EDa59889fa7310b;
+  address public constant IMMUNEFY = 0x7119f398b6C06095c6E8964C1f58e7C1BAa79E18;
```
Comment (eboadom) [inline src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/AaveV3Ethereum_MayFundingUpdatePartB_20240917.sol:25]: Typo, "immunefi"

File: src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/AaveV3Ethereum_MayFundingUpdatePartB_20240917.sol [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  uint256 public constant RECIPIENT_500_AMOUNT = 500e18;
```
Comment (eboadom) [inline src/20240917_AaveV3Ethereum_MayFundingUpdatePartB/AaveV3Ethereum_MayFundingUpdatePartB_20240917.sol:26]: It doesn't really make sense to:
1. Define a constant including the amount in the naming.
2. Assign the constant value.
3. Use the constant below.

Given that you use set the amount on the object of the migrations, better to also hardcode the amount directly on the transfer below. Probably even the recipient too

**Review by sakulstra — APPROVED** [review]

**Pull request merged**